internal/tfr/doc.go

```diff
@@ -0,0 +1,7 @@
+// Package tfr contains functions and routines for interacting with terraform registries using the Module Registry
+// Protocol documented in the official terraform docs:
+// https://www.terraform.io/docs/internals/module-registry-protocol.html
+//
+// MAINTAINER'S NOTE: This package is intentionally set to internal as we do not want to support usage of this outside
```

Comment:
NIT: prob worth mentioning the context from the PR description here: namely, that Terraform's own code is in an `internal` package, so to respect their design, we aren't using it directly, and similarly putting our code in an `internal` package.

Reply:
Done: 95d8610